File: Data Structures & Algorithms/counting-bits/submission-0.cpp
class Solution {
public:
    vector<int> countBits(int n) {
        vector<int> res;
        res.push_back(0);

        int count = 0;
        for (int i = 1; i <= n; i++) {
            int temp = i;
            while (temp > 0) {
                if ((temp & 1) == 1) {
                    count++;
                }
                temp = temp >> 1;
            }
            res.push_back(count);
            count = 0;
        }
        return res;
    }
};

File: Data Structures & Algorithms/daily-temperatures/submission-0.py
class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:
        # for each temp, push a pair (temp, index) onto the stack.
        # if the next element to be pushed onto the stack is greater,
        # compute the difference in indices between them and put the
        # result in the corresponding index in the output array

        res = [0] * len(temperatures)
        stack = []

        for i, temp in enumerate(temperatures):
            while stack and temp > stack[-1][0]:
                stackTemp, stackI = stack.pop()
                res[stackI] = (i - stackI)
            stack.append((temp, i))

        return res